test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -52,17 +54,24 @@ UcxLog::UcxLog(const char* prefix, bool enable) : _enable(enable)
     struct tm tm;
     char str[32];
     if (use_human_time) {
-        strftime(str, sizeof(str), "[%a %b %d %T] ", localtime_r(&tv.tv_sec, &tm));
+        strftime(str, sizeof(str), "[%a %b %d %T ", localtime_r(&tv.tv_sec, &tm));
     } else {
-        snprintf(str, sizeof(str), "[%lu.%06lu] ", tv.tv_sec, tv.tv_usec);
+        snprintf(str, sizeof(str), "[%lu.%06lu ", tv.tv_sec, tv.tv_usec);
     }
-    std::cout << str << prefix << " ";
+
+    size_t str_len = strlen(str);
+    snprintf(str + str_len, sizeof(str) - str_len, "%d] ", getpid());
+
+    _ss = new std::stringstream();
+    (*_ss) << str << prefix << " ";
```

Comment:
can be optimized a bit:
`_ss = new std::stringstream(std::string(str) + prefix + ' ');`

Reply:
why _ss has to be a pointer?

Reply:
@evgeny-leksikov suggested to do this so that in case where 'enable' is false, which is the default for VERBOSE_LOG, then _ss would be smaller.

but it seems to affect perf too (checked on 4 and 8 byte messages):

UCX_SOCKADDR_CM_ENABLE=y ./io_demo -o write -d 8 -i 100000 -w 64 1.1.3.5
UCX_SOCKADDR_CM_ENABLE=y ./io_demo -o read -d 8 -i 100000 -w 64 1.1.3.5
 
_ss is not ptr:

[1610550639.988388 74323] [DEMO] write 1.19643 MB/s min:100000 (1.1.3.5) max:100000 total:100000 ops, active:1, buffers:1

[1610550699.352334 74440] [DEMO] read 1.27935 MB/s min:100000 (1.1.3.5) max:100000 total:100000 ops, active:1, buffers:1

 
_ss is ptr:

[1610550933.986666 74826] [DEMO] write 1.92544 MB/s min:100000 (1.1.3.5) max:100000 total:100000 ops, active:1, buffers:1

[1610550908.074027 74768] [DEMO] read 1.80415 MB/s min:100000 (1.1.3.5) max:100000 total:100000 ops, active:1, buffers:1

'_ss is ptr''s perf  is same as without this PR.

Reply:
maybe use "placement new" http://www.cs.technion.ac.il/users/yechiel/c++-faq/placement-new.html
to avoid extra alloc/release?

Reply:
my point was not to make _ss smaller, need to avoid calling std::stringstream ctor/dtor when it's not needed. IMHO placement new is unjustified here since when verbose mode is enabled we have much more overhead in operator <<
BTW *_ss pointer can replace _enabled flag

Reply:
> can be optimized a bit:
> `_ss = new std::stringstream(std::string(str) + prefix + ' ');`

not addressed

Reply:
after discussing offline, keeping this as is.